src/theory/quantifiers/term_enumeration.cpp

```diff
@@ -0,0 +1,131 @@
+/*********************                                                        */
+/*! \file term_enumeration.cpp
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Andrew Reynolds
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2017 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief Implementation of term enumeration utility
+ **/
+
+#include "theory/quantifiers/term_enumeration.h"
+
+#include "theory/quantifiers/term_util.h"
+#include "theory/rewriter.h"
+
+using namespace CVC4::kind;
+
+namespace CVC4 {
+namespace theory {
+namespace quantifiers {
+
+Node TermEnumeration::getEnumerateTerm(TypeNode tn, unsigned index)
+{
+  Trace("term-db-enum") << "Get enumerate term " << tn << " " << index
+                        << std::endl;
+  std::unordered_map<TypeNode, unsigned, TypeNodeHashFunction>::iterator it =
+      d_typ_enum_map.find(tn);
+  unsigned teIndex;
+  if (it == d_typ_enum_map.end())
+  {
+    teIndex = (int)d_typ_enum.size();
```

Comment:
Use `size_t` for `teIndex`, then you also don't need type casts.

Reply:
this cast was spurious, I'll remove

Reply:
Yes, but you should still use size_t for teIndex since you use it to index d_typ_enum. It's just cleaner.

Reply:
In this case, yes I agree it makes sense to change unsigned -> size_t, since the d_typ_enum_map and teIndex seem to be fully isolated.  I've changed this in the commit.